src/app/clusters/window-covering-server/window-covering-server.cpp

```diff
@@ -0,0 +1,685 @@
+    }
+
+    if (value < minimum)
+    {
+        return 0;
+    }
+
+    if (range > 0)
+    {
+        return static_cast<uint16_t>(WC_PERCENT100THS_MAX * (value - minimum) / range);
+    }
+
+    return WC_PERCENT100THS_MAX;
+}
+
+template <typename ActuatorImpl>
+uint16_t WindowCover::Actuator<ActuatorImpl>::Percent100thsToValue(uint16_t percent100ths)
+{
+    uint16_t openLimit   = OpenLimitGet();
+    uint16_t closedLimit = ClosedLimitGet();
+    uint16_t minimum = 0, maximum = UINT16_MAX, range = UINT16_MAX;
+
+    if (openLimit > closedLimit)
+    {
+        minimum = closedLimit;
+        maximum = openLimit;
+    }
+    else
+    {
+        minimum = openLimit;
+        maximum = closedLimit;
+    }
+
+    range = static_cast<uint16_t>(maximum - minimum);
+
+    if (percent100ths > WC_PERCENT100THS_MAX)
+        return maximum;
+
+    return static_cast<uint16_t>(minimum + ((range * percent100ths) / WC_PERCENT100THS_MAX));
+}
+
+//------------------------------------------------------------------------------
+// LiftActuator
+//------------------------------------------------------------------------------
+
+void WindowCover::LiftActuator::OpenLimitSet(uint16_t limit)
+{
+    writeAttribute(mEndPoint, ZCL_WC_INSTALLED_OPEN_LIMIT_LIFT_ATTRIBUTE_ID, static_cast<uint8_t *> &limit, ZCL_INT16U_ATTRIBUTE_TYPE);
+}
+
+uint16_t WindowCover::LiftActuator::OpenLimitGet(void)
+{
+    uint16_t limit = 0;
+    readAttribute(mEndPoint, ZCL_WC_INSTALLED_OPEN_LIMIT_LIFT_ATTRIBUTE_ID, static_cast<uint8_t *> &limit, sizeof(uint16_t));
+    return limit;
+}
+
+void WindowCover::LiftActuator::ClosedLimitSet(uint16_t limit)
+{
+    writeAttribute(mEndPoint, ZCL_WC_INSTALLED_CLOSED_LIMIT_LIFT_ATTRIBUTE_ID, static_cast<uint8_t *> &limit, ZCL_INT16U_ATTRIBUTE_TYPE);
+}
+
+uint16_t WindowCover::LiftActuator::ClosedLimitGet(void)
+{
+    uint16_t limit = 0;
+    readAttribute(mEndPoint, ZCL_WC_INSTALLED_CLOSED_LIMIT_LIFT_ATTRIBUTE_ID, static_cast<uint8_t *> &limit, sizeof(uint16_t));
+    return limit;
+}
+
+void WindowCover::LiftActuator::PositionSet(uint16_t percent100ths)
+{
+    uint8_t percent = static_cast<uint8_t>(percent100ths / 100);
+    uint16_t value  = Percent100thsToValue(percent100ths);
+
+    writeAttribute(mEndPoint, ZCL_WC_CURRENT_POSITION_LIFT_ATTRIBUTE_ID, static_cast<uint8_t *> &value, ZCL_INT16U_ATTRIBUTE_TYPE);
+    writeAttribute(mEndPoint, ZCL_WC_CURRENT_POSITION_LIFT_PERCENTAGE_ATTRIBUTE_ID, static_cast<uint8_t *> &percent, ZCL_INT8U_ATTRIBUTE_TYPE);
+    writeAttribute(mEndPoint, ZCL_WC_CURRENT_POSITION_LIFT_PERCENT100_THS_ATTRIBUTE_ID, static_cast<uint8_t *> &percent100ths,
+                   ZCL_INT16U_ATTRIBUTE_TYPE);
+    emberAfWindowCoveringClusterPrint("LiftActuator::PositionSet(%u%%)", percent100ths / 100);
+}
+
+uint16_t WindowCover::LiftActuator::PositionGet()
+{
+    uint16_t percent100ths = 0;
+    readAttribute(mEndPoint, ZCL_WC_CURRENT_POSITION_LIFT_PERCENT100_THS_ATTRIBUTE_ID, static_cast<uint8_t *> &percent100ths,
```

Comment:
```suggestion
    readAttribute(mEndPoint, ZCL_WC_CURRENT_POSITION_LIFT_PERCENT100_THS_ATTRIBUTE_ID, static_cast<uint8_t *>(&percent100ths),
```

Reply:
Replaced with the corresponding Accessors.h function.